src/mpid/ch3/channels/nemesis/src/mpid_nem_init.c

```diff
@@ -409,6 +409,8 @@ MPID_nem_init(int pg_rank, MPIDI_PG_t *pg_p, int has_parent ATTRIBUTE((unused)))
 
     mpi_errno = MPIDI_CH3_SHM_Init();
     MPIR_ERR_CHECK(mpi_errno);
+    mpi_errno = MPIDU_Init_shm_barrier();
+    MPIR_ERR_CHECK(mpi_errno);
```

Comment:
what's the reasoning for adding another shm barrier here?

Reply:
We need a barrier before `MPIDU_Init_shm_finalize`. In ch3, all init_shm usages are in MPI_Init. Having a shm barrier here ensures processes won't enter finalize when init_shm is still in use.